libraries/wallet/wallet.cpp

```diff
@@ -608,9 +608,16 @@ class wallet_api_impl
    {
       return get_account(account_name_or_id).get_id();
    }
+   std::string asset_id_to_string(asset_id_type id) const
+   {
+      std::string asset_id = fc::to_string(id.space_id) +
+                             "." + fc::to_string(id.type_id) +
+                             "." + fc::to_string(id.instance.value);
+      return asset_id;
+   }
    optional<asset_object> find_asset(asset_id_type id)const
    {
-      auto rec = _remote_db->get_assets({id}).front();
+      auto rec = _remote_db->get_assets({asset_id_to_string(id)}).front();
```

Comment:
This looks ugly as well.

Perhaps better to add another `get_assets_by_id()` API?

Reply:
Why didn't change `find_asset(asset_id_type id)` to `find(string)`? Is it an function that only used internally?

Reply:
Anyway, I think cli_wallet is less important, it's fine to have some ugly code and slightly reduced performance.